crates/turbopack-ecmascript/src/chunk/content.rs

```diff
@@ -213,110 +224,35 @@ impl EcmascriptChunkContentVc {
-        writeln!(code, "{}, {{", stringify_js(chunk_server_path))?;
+        writedoc!(
+            code,
+            r#"
+                (self.TURBOPACK = self.TURBOPACK || []).push([
+                {chunk_path},
+                {{
+            "#,
+            chunk_path = StringifyJs::new(chunk_server_path)
+        )?;
+
         for entry in &this.module_factories {
-            write!(code, "\n{}: ", &stringify_js(entry.id()))?;
+            write!(code, "\n{}: ", StringifyJs::new(entry.id()))?;
             code.push_code(entry.code());
-            code += ",";
+            write!(code, ",")?;
         }
-        code += "\n}";
-
-        if let Some(evaluate) = &this.evaluate {
-            let evaluate = evaluate.await?;
-            let condition = evaluate
-                .ecma_chunks_server_paths
-                .iter()
-                .map(|path| format!(" && loadedChunks.has({})", stringify_js(path)))
-                .collect::<Vec<_>>()
-                .join("");
-            let entries_instantiations = evaluate
-                .entry_modules_ids
-                .iter()
-                .map(|id| async move {
-                    let id = id.await?;
-                    let id = stringify_js(&id);
-                    Ok(format!(r#"    instantiateRuntimeModule({id});"#)) as Result<_>
-                })
-                .try_join()
-                .await?
-                .join("\n");
-
-            let chunk_list_register = evaluate
-                .chunk_list_path
-                .as_deref()
-                .map(|path| {
-                    format!(
-                        r#"registerChunkList({}, {});"#,
-                        stringify_js(&path),
-                        stringify_js(
-                            &evaluate
-                                .ecma_chunks_server_paths
-                                .iter()
-                                .chain(&evaluate.other_chunks_server_paths)
-                                .collect::<Vec<_>>()
-                        )
-                    )
-                })
-                .unwrap_or_else(String::new);
-            // Add a runnable to the chunk that requests the entry module to ensure it gets
-            // executed when the chunk is evaluated.
-            // The condition stops the entry module from being executed while chunks it
-            // depend on have not yet been registered.
-            // The runnable will run every time a new chunk is `.push`ed to TURBOPACK, until
-            // all dependent chunks have been evaluated.
-            writedoc!(
-                code,
-                r#"
-                    , ({{ loadedChunks, instantiateRuntimeModule, registerChunkList }}) => {{
-                        if (!(true{condition})) return true;
-                        {chunk_list_register}
-                    {entries_instantiations}
-                    }}
-                "#
-            )?;
+
+        write!(code, "\n}}")?;
```

Comment:
This should be controlled by the runtime.

Other runtimes might not use a global var to communicate between chunks.

e.g. they could use ESM `export` and `import()` to load chunks, or in node.js they could use `module.modules = ...` and `require()` to load chunks.

Reply:
Agreed. This is why I'm also trying to rework our `EcmascriptChunkEntry` system to make it so it can be passed to the runtime.